app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HomeDatabaseViewModel.kt

```
@@ -1,9 +1,221 @@
+            }
+            
+            // Create habit rows with real progress
+            val items = habitDtos.map { habit ->
+                val habitCheckins = allCheckins.filter { it.habitId == habit.id }
+                val today = LocalDate.now()
+                val lastThreeDays = getLastThreeDays()
+                val completedLastThreeDays = habitCheckins.mapNotNull { checkin ->
+                    try {
+                        // Parse the dayKey directly since it's in yyyy-MM-dd format
+                        java.time.LocalDate.parse(checkin.dayKey)
+                    } catch (e: Exception) { 
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)